exercises/binary-search/binary_search.exs

```diff
@@ -0,0 +1,24 @@
+defmodule BinarySearch do
+  @doc """
+    Searches for a key in the list using the binary search algorithm.
+    It returns :not_found if the key is not in the list.
+    Otherwise returns the tuple {:ok, index}.
+
+    ## Examples
+
+      iex> BinarySearch.search([], 2)
+      :not_found
+
+      iex> BinarySearch.search([1, 3, 5], 2)
+      :not_found
+
+      iex> BinarySearch.search([1, 3, 5], 5)
+      {:ok, 2}
+
+  """
+
+  @spec search(Enumerable.t, integer) :: {:ok, integer} | :not_found
+  def search([], key) do
```

Comment:
probably make this first argument a variable name, leaving implementation up to the solver